src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -287,4 +253,53 @@ private static bool IsWebProject(ProjectResource projectResource)
         var launchProfile = projectResource.GetEffectiveLaunchProfile(throwIfNotFound: true);
         return launchProfile?.ApplicationUrl != null;
     }
+
+    private static void SetAspNetCoreUrls(this IResourceBuilder<ProjectResource> builder)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return;
+        }
+
+        builder.WithEnvironment(context =>
+        {
+            if (context.EnvironmentVariables.ContainsKey("ASPNETCORE_URLS"))
+            {
+                // If the user has already set ASPNETCORE_URLS, we don't want to override it.
+                return;
+            }
+
+            var aspnetCoreUrls = new ReferenceExpressionBuilder();
+
+            var processedHttpsPort = false;
+            var first = true;
+
+            static bool IsValidAspNetCoreUrl(EndpointAnnotation e) =>
+                e.UriScheme is "http" or "https" && e.TargetPortEnvironmentVariable is null;
+
+            // Turn http and https endpoints into a single ASPNETCORE_URLS environment variable.
+            foreach (var e in builder.Resource.GetEndpoints().Where(e => IsValidAspNetCoreUrl(e.EndpointAnnotation)))
+            {
+                if (!first)
+                {
+                    aspnetCoreUrls.AppendLiteral(";");
+                }
+
+                if (!processedHttpsPort && e.EndpointAnnotation.UriScheme == "https")
+                {
+                    // Add the environment variable for the HTTPS port if we have an HTTPS service. This will make sure the
+                    // HTTPS redirection middleware avoids redirecting to the internal port.
+                    context.EnvironmentVariables["ASPNETCORE_HTTPS_PORT"] = e.Property(EndpointProperty.Port);
+
+                    processedHttpsPort = true;
+                }
+
+                aspnetCoreUrls.Append($"{e.Property(EndpointProperty.Scheme)}://localhost:{e.Property(EndpointProperty.TargetPort)}");
+                first = false;
+            }
+
+            // Combine into a single expression
+            context.EnvironmentVariables["ASPNETCORE_URLS"] = aspnetCoreUrls.Build();
```

Comment:
If this is empty, is it better to set it or not set it?

Reply:
Fixed